File: backend/app/schemas/meal.py
from __future__ import annotations

from decimal import Decimal

from pydantic import BaseModel


class MealDto(BaseModel):
    id: int | None = None
    restaurantId: int | None = None
    name: str | None = None
    description: str | None = None
    weight: int | None = None
    calorie: int | None = None
    imageLink: str | None = None
    category: str | None = None
    price: Decimal | None = None
    available: bool | None = None


class MealCategoryAvailabilityPatchDto(BaseModel):
    restaurantId: int
    category: str
    available: bool

File: user_panel/src/types/meal.ts
export interface Meal {
  id: number;
  restaurant_id: number;
  name: string;
  description: string | null;
  weight: number | null;
  calorie: number | null;
  image_link: string | null;
  category: string | null;
  price: number;
  is_available: boolean;
}
